pallets/dkg-proposal-handler/src/lib.rs

```diff
@@ -365,58 +371,65 @@ impl<T: Config> Pallet<T> {
 	// *** Offchain worker methods ***
 
 	fn submit_signed_proposal_onchain(block_number: T::BlockNumber) -> Result<(), &'static str> {
-		let signer = Signer::<T, <T as Config>::OffChainAuthId>::all_accounts();
-		if !signer.can_sign() {
-			return Err(
-				"No local accounts available. Consider adding one via `author_insertKey` RPC.",
-			)?
-		}
-		match Self::get_next_offchain_signed_proposal(block_number) {
-			Ok(next_proposals) => {
-				// We filter out all proposals that are already on chain
-				let filtered_proposals = next_proposals
-					.iter()
-					.cloned()
-					.filter(Self::is_existing_proposal)
-					.collect::<Vec<_>>();
-
-				// We split the vector into chunks of `T::MaxSubmissionsPerBatch` length and submit
-				// those chunks
-				for chunk in filtered_proposals.chunks(T::MaxSubmissionsPerBatch::get() as usize) {
-					let call = Call::<T>::submit_signed_proposals { props: chunk.to_vec() };
-					let result = signer
-						.send_signed_transaction(|_| call.clone())
-						.into_iter()
-						.map(|(_, r)| r)
-						.collect::<Result<Vec<_>, _>>()
-						.map_err(|()| "Unable to submit unsigned transaction.");
-					// Display error if the signed tx fails.
-					if result.is_err() {
-						frame_support::log::error!(
-							target: "dkg_proposal_handler",
-							"failure: failed to send unsigned transactiion to chain: {:?}",
-							call,
-						);
-					} else {
-						// log the result of the transaction submission
-						frame_support::log::debug!(
-							target: "dkg_proposal_handler",
-							"Submitted unsigned transaction for signed proposal: {:?}",
-							call,
-						);
+		let mut lock = StorageLock::<Time>::new(b"submit_signed_proposal_onchain::lock");
```

Comment:
And Also here:
> Could be these keys here `b"submit_genesis_public_key_onchain::lock"` be constants, similar to `AGGREGATED_PUBLIC_KEYS_AT_GENESIS` for example? To avoid typos.